data/api/client-server/directory.yaml

```diff
@@ -61,6 +61,15 @@ paths:
             application/json: {}
           schema:
             type: object
+        400:
+          description: The given `roomAlias` is not a valid room alias.
```

Comment:
"Define valid". We can probably copy a lot of the text from https://github.com/matrix-org/matrix-spec/pull/1261 and use it here.

Generally though, I'm more in favour of fixing the spec to say that 400 `M_INVALID_PARAM` is something which can appear on every endpoint rather than each endpoint individually. This is likely a larger effort than this PR should take on, however.

Reply:
I would ideally link to https://spec.matrix.org/latest/appendices/#room-aliases here, but I couldn't find an example of such a hyperlink from the OpenAPI spec.

Reply:
There is https://github.com/matrix-org/matrix-spec/blob/dcc1f4adee2be41069045ddbf3ef913e9c19d116/data/api/client-server/definitions/one_time_keys.yaml#L21 but this a) is hardcoded to a spec version and b) doesn't appear in the spec (#1287)

Reply:
Oh, directly above this:

https://github.com/matrix-org/matrix-spec/blob/dcc1f4adee2be41069045ddbf3ef913e9c19d116/data/api/client-server/definitions/one_time_keys.yaml#L7

That'll do.

Reply:
> There is
> 
> https://github.com/matrix-org/matrix-spec/blob/dcc1f4adee2be41069045ddbf3ef913e9c19d116/data/api/client-server/definitions/one_time_keys.yaml#L21
> 
> but this a) is hardcoded to a spec version and b) doesn't appear in the spec (#1287)

This looks like a bug, ftr. Will take a look.